api/src/main/kotlin/com/openlattice/admin/AdminApi.kt

```diff
@@ -101,4 +106,51 @@ interface AdminApi {
 
     @PATCH(BASE + JOBS + ID_PATH)
     fun updateJob(@Path(ID) jobId: UUID, @Body update: JobUpdate): Map<UUID, DistributableJob<*>>
+
+    /**
+     *  Gets all data warehouses known by the application.
+     *
+     *  @return List containing JDBC connection details for all known warehouses.
+     */
+
+    @GET(BASE + WAREHOUSES)
+    fun getWarehouses(): Iterable<JdbcConnectionParameters>
+
+    /**
+     *  Gets JDBC connection details for a specified Warehouse
+     *
+     *   @param WarehouseId the ID for the warehouse to get details about.
+     *   @return The JDBC connection details that define the warehouse.
+     *   - Fields: Title, Url, Driver, Database, Username, Password, Properties, Description
+     *
+     */
+    @GET(BASE + WAREHOUSES + WAREHOUSE_ID_PATH)
+    fun getWarehouse(@Path(WAREHOUSE_ID_PARAM) WarehouseId: UUID): JdbcConnectionParameters
+
+    /**
+     *  Creates a warehouse by providing the application JDBC details needed
+     *  to connect.
+     *
+     *  @param JdbcConnection The JDBC connection details that define the warehouse.
+     *  - Fields: Title, Url, Driver, Database, Username, Password, Properties, Description
+     *   @return The ID of the new warehouse.
+     */
+    @POST(BASE + WAREHOUSES)
+    fun createWarehouse(@Body jdbc: JdbcConnectionParameters): UUID
+
+    /**
+     *  Deletes a specified warehouse from the known warehouse connections.
+     *
+     *  @param WarehouseID The ID of the warehouse to be deleted.
+     */
+    @DELETE(BASE + WAREHOUSES + WAREHOUSE_ID_PATH)
+    fun deleteWarehouse(@Path(WAREHOUSE_ID_PARAM) WarehouseId: UUID)
+
+    /**
+     * Update the connection details for a warehouse JDBC connection.
+     *
+     *  @param jdbc JdbcConnectionParameters to replace existing parameters. Jdbc.id indicates the Warehouse to be updated.
+     */
+    @PATCH(BASE + WAREHOUSES)
+    fun updateWarehouse(@Body jdbc: JdbcConnectionParameters): Int
```

Comment:
If I only want to update a single param (or just a few), do I still need to pass the entire object with every param filled in? If so, that's not ideal.

Reply:
Yes, that's how it's implemented now. Will add endpoints for individual params.